experiments/yarn/src/main/scala/org/apache/gearpump/experiments/yarn/master/AmActor.scala

```diff
@@ -97,19 +141,15 @@ class AmActor(appConfig: AppConfig, yarnConf: YarnConfiguration) extends Actor {
       
     case containerRequest: ContainerRequestMessage =>
       LOG.info("AM: Received ContainerRequestMessage")
-      amRMClient ! containerRequest
+      rmClientActor ! containerRequest
```

Comment:
This is early drawing of message exchange (it may differ now).
![yarnam mermaid](https://cloud.githubusercontent.com/assets/146453/7627482/af1a6268-f9cc-11e4-9ef2-aa93507b6551.png)
Different actor creation is to eliminate Thread manipulation (if you look at many non trivial AppMasters for example [spark](https://github.com/freedafeng/adspark/blob/bebd1924a9b434fcb4fa719f515fe910fd663ba2/yarn/stable/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala) you'll see many Thread specific operations. The goal was to have actors exchange messages to do this work rather than do it within different Thread contexts.